### PR TITLE
Enable AWS SQS JSON Preview

### PR DESCRIPTION
**Pull request by ziyanli-amazon**
Enable AWS SQS JSON Preview
ref: https://code.amazon.com/packages/AwsSdkGo-SqsJsonPreview/trees/mainline

### CODE REVIEW

File: service/sqs/service.go [hunk 1]
```diff
@@ -8,7 +8,8 @@ import (
 	"github.com/aws/aws-sdk-go/aws/client/metadata"
 	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/aws/signer/v4"
-	"github.com/aws/aws-sdk-go/private/protocol/query"
+	"github.com/aws/aws-sdk-go/private/protocol"
+	"github.com/aws/aws-sdk-go/private/protocol/jsonrpc"
 )
```
Comment (eddy-aws) [inline service/sqs/service.go:13]: Can we add sqs specific unit tests? Mock raw json response and make sure a valid error code is returned.

Comment (ziyanli-amazon) [reply service/sqs/service.go:13]: Sure. Will add some unit tests to validate the error codes.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# AWS SDK for Go
+# AWS SDK for Go (with modified SQS API that uses JSON wire protocol)
```
Comment (greensopinion) [inline README.md:1]: Is the intention to have this PR merged upstream? If so, consider removing this note.

File: example/service/sqs/mockingClientsForTests/errorHandlingExample.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
+//go:build example
```
Comment (greensopinion) [inline example/service/sqs/mockingClientsForTests/errorHandlingExample.go:1]: if this example is sqs-specific, would it make sense to name the file that way? e.g. `sqsExample.go` or `sqsErrorHandlingExample.go`?

File: example/service/sqs/mockingClientsForTests/errorHandlingExample.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+const SQS_ENDPOINT_VAR = "SQS_ENDPOINT"
+
+var OriginalMessage = "a-message"
+
+var OriginalMessageMd5 = md5String(OriginalMessage)
+
+func CreateSession() (*session.Session, error) {
+	return session.NewSession(&aws.Config{
+		Endpoint: aws.String(os.Getenv(SQS_ENDPOINT_VAR)),
```
Comment (greensopinion) [inline example/service/sqs/mockingClientsForTests/errorHandlingExample.go:24]: how does this behave if `SQS_ENDPOINT_VAR` is not provided?

File: private/model/api/api.go [hunk 5]
```diff
@@ -254,6 +255,13 @@ func (a *API) ShapeListErrors() []*Shape {
 	return list
 }
 
+func (a *API) AwsQueryCompatibleErrorCode(errorCode string) string {
+	if len(a.AwsQueryCompatible) > 0 && a.AwsQueryCompatible[errorCode] != nil && a.AwsQueryCompatible[errorCode]["code"] != "" {
+		return a.AwsQueryCompatible[errorCode]["code"]
```
Comment (greensopinion) [inline private/model/api/api.go:260]: would it make sense to have local variables to avoid repeated map lookups?

**Review by greensopinion — COMMENTED** [review]
Looks great! A few minor suggestions and questions inline.

File: service/sqs/service.go (hunk 1 above)
Comment (eddy-aws) [reply service/sqs/service.go:13]: I see that you have added examples with the integration tests. How do we know if they are run? What target should be invoked to run them as integration tests? And, why do they have "mockClients" in their path? What exactly is being mocked?

File: README.md (hunk 2 above)
Comment (ziyanli-amazon) [reply README.md:1]: No, I won't be able to merge upstream due to permission. This PR is intended for customers to try out the private Beta prior to JSON GA.

File: example/service/sqs/mockingClientsForTests/errorHandlingExample.go (hunk 3 above)
Comment (ziyanli-amazon) [reply example/service/sqs/mockingClientsForTests/errorHandlingExample.go:1]: There's already one example in the `sqs` directory, I'm not sure if `sqs` prefix is still necessary but i can certainly add it.

**Pull request closed without merging**